fashion-messaging-app-api/server.js

```diff
@@ -133,17 +137,28 @@ app.get('/photos', async (req, res) => {
 
     const fashionPhotos = await Promise.all(fetchPromises);
     const validPhotos = fashionPhotos.filter(photo => photo !== undefined && photo !== null);
-
     for (let photo of validPhotos) {
-      await SearchResult.create({
-        searchTerm: query,
-        imageUrl: photo.src.original,
-        isFashion: true,  
-        userId: req.session.user ? req.session.user.id : null,  
+      let searchResult = await SearchResult.findOne({
+        where: { 
+          searchTerm: query,
+          userId: req.session.user ? req.session.user.id : null,
+        },
       });
+    
+      if (searchResult) {
+        searchResult.searchCount += 1;
+        await searchResult.save();
+      } else {
+        await SearchResult.create({
+          searchTerm: query,
+          imageUrl: photo.src.original,
+          isFashion: true,  
+          userId: req.session.user ? req.session.user.id : null,  
+          searchCount: 1,
+        });
+      }
     }
```

Comment:
This checks if the searchResult has been stored in my database. If it has, then I want to increase the count then save how many times the user has searched for that outfit

Reply:
this looks great!